src/dfm/setup.rs

```diff
@@ -0,0 +1,25 @@
+use super::model::Config;
+use anyhow::{Result, anyhow};
+use std::fs;
+use std::path::PathBuf;
+
+pub fn setup(path: PathBuf) -> Result<()> {
+    let config = generate_config(path)
+
+    println!("{:?}", config?);
+
+    Ok(())
+}
+
+fn generate_config(path: PathBuf) -> Result<Config> {
+    let config_content: String = fs::read_to_string(&path).expect("");
+
+    let extension = path.extension().expect("").to_str().unwrap();
+
+    match extension {
+        "yml" | "yaml" => Ok(serde_yaml::from_str(&config_content).unwrap()),
+        "json" | "json5" => Ok(serde_json::from_str(&config_content).unwrap()),
+        "toml" => Ok(toml::from_str(&config_content).unwrap()),
```

Comment:
```suggestion
        "yml" | "yaml" => Ok(serde_yaml::from_str(&config_content)?),
        "json" | "json5" => Ok(serde_json::from_str(&config_content)?),
        "toml" => Ok(toml::from_str(&config_content)?),
```

Unwraps panic. Just rethrow the error into your Result type. It's fine. Alternatively, you can add context -- however i think serde_json is usually pretty clear about the error.